lib/services/devices/deviceRegistryMongoDB.js

```diff
@@ -66,6 +67,11 @@ function storeDevice(newDevice, callback) {
         deviceObj[attributeList[i]] = newDevice[attributeList[i]];
     }
 
+    // Ensure protocol is in json
```

Comment:
This is generic code (in the library) independent from transport (JSON, UL or whatever... so this comment text is a bit surprising. Maybe the following is more accurate?

```
// If protocol is missing, then use default one from configuration
```

althougth I could be interpreting incorrectly the fix in this PR

Reply:
protocol, is expected to be in newDevice object